server/plugin/plugin.go

```diff
@@ -250,6 +263,86 @@ func (p *Plugin) OnDeactivate() error {
 	return nil
 }
 
+func (p *Plugin) ReactionHasBeenAdded(c *plugin.Context, reaction *model.Reaction) {
+	githubEmoji := p.emojiMap[reaction.EmojiName]
+	if githubEmoji == "" {
+		return
+	}
+	post, error := p.client.Post.GetPost(reaction.PostId)
+	if error != nil {
+		p.API.LogError("Invalid post id. Addition of reaction failed.")
+		return
+	}
+
+	repo := post.GetProp("gh_repo")
+	if repo == nil {
+		p.API.LogDebug("Required prop is not present. This post is not created by this plugin, hence ignoring")
+		return
+	}
+
+	orgRepo := strings.Split(repo.(string), "/")
+	// strange that gh_object_id is always returned as float64, without type casting it throws an error
+	id := int(post.GetProp("gh_object_id").(float64))
+
+	info, appErr := p.getGitHubUserInfo(reaction.UserId)
+	if appErr != nil {
+		if appErr.ID != apiErrorIDNotConnected {
+			p.API.LogError("Error in getting user info", "error", appErr.Message)
+		}
+	}
+	ghClient := p.githubConnectUser(context.Background(), info)
+
+	_, _, error = ghClient.Reactions.CreateIssueCommentReaction(context.Background(), orgRepo[0], orgRepo[1], int64(id), githubEmoji)
+	if error != nil {
+		p.API.LogError("Error found while creating issue comment reaction", error)
+	}
+}
+
+func (p *Plugin) ReactionHasBeenRemoved(c *plugin.Context, reaction *model.Reaction) {
+	githubEmoji := p.emojiMap[reaction.EmojiName]
+	if githubEmoji == "" {
+		return
+	}
+	post, error := p.client.Post.GetPost(reaction.PostId)
+	if error != nil {
+		p.API.LogError("Invalid post id. Removal of reaction failed.")
+		return
+	}
+
+	repo := post.GetProp("gh_repo")
+	if repo == nil {
+		p.API.LogDebug("Required prop is not present. This post is not created by this plugin, hence ignoring")
+		return
+	}
+
+	orgRepo := strings.Split(repo.(string), "/")
+	// strange that gh_object_id is always returned as float64, without type casting it throws an error
+	id := int(post.GetProp("gh_object_id").(float64))
+
+	info, appErr := p.getGitHubUserInfo(reaction.UserId)
+	if appErr != nil {
+		if appErr.ID != apiErrorIDNotConnected {
+			p.API.LogError("Error in getting user info", "error", appErr.Message)
+		}
+	}
+	ghClient := p.githubConnectUser(context.Background(), info)
+
+	reactions, _, error := ghClient.Reactions.ListIssueCommentReactions(context.Background(), orgRepo[0], orgRepo[1], int64(id), &github.ListOptions{})
+
+	if error == nil {
```

Comment:
We should "return early" instead of having this indented block for the rest of the function

```suggestion
	if err != nil {
	    p.API.LogDebug(...)
	    return
```